src/hybrid_reader.c

```diff
@@ -336,10 +347,8 @@ static void HR_Rewind(void *ctx) {
   HybridIterator *hr = ctx;
   hr->resultsPrepared = false;
   hr->numIterations = 0;
-  if (hr->list) {
-    VecSimQueryResult_Free(hr->list);
-    hr->list = NULL;
-  }
+  VecSimQueryResult_Free(hr->list);
+  hr->list.results = NULL;
```

Comment:
move `hr->list.results = NULL;` into `VecSimQueryResult_Free`

Reply:
OK. I decided not to do that because no other `free` function does that

Reply:
`results` is an internal property of `hr->list`. Why NULL outside of free function?